src/helpers/routes.js

```diff
@@ -85,8 +85,10 @@ exports.makeRoutes = (baseRoutes, {
       const shouldAddPrefix = (
         // No prefix if app uses different locale domains
         !differentDomains &&
-        // Only add prefix on top level routes
-        !isChild &&
+        // Only add prefix when
+        // 1. on top level routes
+        // 2. parsePages is false, since we are not going from parent path to child path
+        !(isChild && parsePages) &&
```

Comment:
Do we really want to skip adding a prefix on children routes in this case? I'm not sure I understand this additional check.

Reply:
This is when it's top level or when we are not parsing pages, we add the prefix.

`!(isChild && parsePages)` is the same as `!child || !parsePages`.

Reply:
This condition is incorrect, pls see the latest commit.